client/admin-pointers.js

```diff
@@ -22,17 +20,29 @@ jQuery( document ).ready( function( $ ) {
 			return;
 		}
 
+		const target = $( pointer.target );
+
 		const options = $.extend( pointer.options, {
 			close: function() {
 				$.post( ajaxurl, {
 					pointer: pointer.id,
 					action: 'dismiss-wp-pointer',
 				} );
+
+				if ( pointer.dim ) {
+					$( '#wcs-pointer-page-dimmer' ).fadeOut( 500 );
+				}
+
 				show_pointer( pointers, i + 1 );
 			},
 		} );
 
-		$( pointer.target ).pointer( options ).pointer( 'open' );
+		target.pointer( options ).pointer( 'open' );
+		if ( pointer.dim ) {
+			target.css( 'z-index', 9999 );
+			$( 'body' ).append( '<div id="wcs-pointer-page-dimmer" class="wcs-pointer-page-dimmer"></div>' );
```

Comment:
What do you think about always rendering the overlay div, but having it's default style be hidden? That way we don't have to manipulated the DOM to add the overlay.

Also - I noticed that the overlay still allows the page to be scrolled behind it. Can we add a class to the body for "dim" pointers that sets `overflow: hidden` to prevent scrolling?